### ISSUE TITLE
Status icon has visible border instead of depending on background

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud-libraries/nextcloud-vue#4873]
The status icon border/background currently is always set to color-main-background. This means that in light theme, it is always white. On other backgrounds like in the header or in various states like hovered/focused/active/selected it looks off.
Just like with other icons such as the favorite overlay in the files list, it is intended to create a slight gap between the icon and the element it overlays – not to be a visible border.

This occurs in various places, here are some examples:

## Top right account menu
The status of the top right avatar. Granted this depends on the theming background, but we can at least solve it for the default installation, and possibly even for the shipped themes.
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/a4857628-3912-4ad9-ad56-1f120c5ccdf3)

## File sharing sidebar
The selected suggestion has a grey background, but is not the same for the status border. Additionally, the status icon seems to be cut off a bit, leading to the crescent shape being barely recognizable?
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/1c1d1143-53d5-4fa5-aa77-020693741c4b)

## Talk navigation
In the default state it is a bit difficult to set a border which fits the background since we have the glass look and the variable background. But in hover/focus and active state we know what the background is.
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/6be6e5a1-9a3f-42c6-a11e-a4766372e0e2)
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/704572cf-5728-4c5c-8597-6c9546ad23f1)
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/edcabedc-6b23-415d-afa9-696f99fb8db7)

## Talk @-mention suggestions
The status icon of the active/selected person should be adjusted to get a border that matches the background of the selected entry.
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/f6b1fff3-5f69-4bc0-a9ac-adf77312bc63)

## Deck assignments
Similar as in other examples, the status for the selected entry would look nicer with the color adjusted. Additionally, in the input field the same issue as in Files happens where the status icon is somehow cut off:
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/24b1b43d-7845-42e4-8fb1-82c47e2e042c)
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/925062/6d0692ad-791e-4342-9e7e-367c5ac892dd)

**Comment by jancborchardt** [comment]
(The concept of the issue is basically the same as for the favorite overlay icon: https://github.com/nextcloud-libraries/nextcloud-vue/issues/4872 )

**Comment by jancborchardt** [comment]
This would be a nice papercut for us to fix in the components? @szaimen are you interested in looking into this?

**Comment by szaimen** [comment]
> This would be a nice papercut for us to fix in the components? @szaimen are you interested in looking into this?

I can look into it next week earliest (i am on vacation tomorrow for one day)

**Comment by susnux** [comment]
@jancborchardt how should this work for the one in the top navigation?
Because if a background image is used it could probably look quite weird.

**Comment by jancborchardt** [comment]
@susnux yeah, that's the case where I'm not really sure what to do. Maybe using primary or a secondary would already be better than white?

Alternatively, there we could use no border at all, as no gap to the avatar might look better than a visible circle around.

Or we use the same color that is used for the avatar border now?
